pkg/login/ldap.go

```diff
@@ -260,9 +260,18 @@ func (a *ldapAuther) SyncOrgRoles(user *m.User, ldapUser *LdapUserInfo) error {
 				match = true
 				if org.Role != group.OrgRole {
 					// update role
-					cmd := m.UpdateOrgUserCommand{OrgId: org.OrgId, UserId: user.Id, Role: group.OrgRole}
-					if err := bus.Dispatch(&cmd); err != nil {
-						return err
+					if setting.LdapAllowRoleSync {
+						// allow sync of org roles
+						cmd := m.UpdateOrgUserCommand{OrgId: org.OrgId, UserId: user.Id, Role: group.OrgRole}
+						if err := bus.Dispatch(&cmd); err != nil {
+							return err
+						}
+					} else {
+						// disallow sync of org roles
+						cmd := m.UpdateOrgUserCommand{OrgId: org.OrgId, UserId: user.Id, Role: org.Role}
```

Comment:
why is this called when LdapAllowRoleSync is disabled? Should it not just return (do nothing) ?